applications/asset_tracker_v2/doc/gnss_module.rst

```diff
@@ -0,0 +1,126 @@
+The module has Kconfig options to control the LNA using either the nRF9160's COEX0 or MAGPIO interafce.
+See :ref:`configuration_options` for more details on how to configure the antenna and LNA.
+
+GPS assistance data
+===================
+
+The GNSS module receives requests for GPS assistance data from the nRF9160 modem.
+When an A-GPS request is received, the module distributes it to the other modules in the form of a ``GNSS_EVT_AGPS_NEEDED`` event containing information about what assistance data is needed.
+Providing the requested A-GPS data may reduce the time it takes to acquire a position fix.
+
+The module can also store the last known location if :kconfig:`CONFIG_GNSS_MODULE_PGPS_STORE_LOCATION` is enabled to improve time to fix when predicted GPS assistance data is used.
+
+Module internals
+================
+
+The GNSS module is a thread-less module in terms of message processing.
+However, the GNSS module differs from the other modules at this point.
+It has a thread dedicated to processing GNSS events from the modem library, as opposed to processing messages like other modules with threads do.
+The thread is used to offload data processing from the GNSS callbacks coming from the modem library, as these are called from interrupt context and no time consuming tasks should be performed there.
+
+All incoming events from other modules are handled in the context of the event manager callback, as they all complete fast enough to not require a separate thread.
+
+.. _configuration_options:
+
+Configuration options
+*********************
+
+To enable the GNSS module, you must set the following option:
+
+ * :kconfig:`CONFIG_GNSS_MODULE`
+
+The module can report data either as a structure of position, velocity and time (PVT) information, or as an NMEA string of GGA type.
+To select which format to use, configure on of the two options below:
+
+ * :kconfig:`CONFIG_GNSS_MODULE_PVT` - Include position, velocity and time (PVT) information in GNSS_EVT_DATA_READY events that are emitted when the GNSS acquires a position fix.
+
+ * :kconfig:`CONFIG_GNSS_MODULE_NMEA` - Include NMEA string of GGA type in GNSS_EVT_DATA_READY events that are emitted when the GNSS acquires a position fix.
+
+The GNSS module can set up the modem and LNA to use either onboard or external GNSS antenna using on of the two options below:
+
+ * :kconfig:`CONFIG_GNSS_MODULE_ANTENNA_ONBOARD` - Send antenna configurations to the modem when you compile for Thingy:91 or nRF9160 DK.
+
+ * :kconfig:`CONFIG_GNSS_MODULE_ANTENNA_EXTERNAL` - This will configure the target for using an external GNSS antenna.
+
+If P-GPS is used, the last known position from a GNSS fix can be stored and injected to the modem together with the relevant ephemeris.
+This may, depending on the use-case and the device's movements, reduce the time to fix.
+To enable this feature, use the option below:
```

Comment:
```suggestion
This may, depending on the use case and the device's movements, reduce the time to fix.
To enable this feature, use the following option:
```